src/docker/shared.rs

```diff
@@ -508,31 +515,47 @@ pub(crate) fn custom_image_build(
                 )
                 .wrap_err("when pre-building")
                 .with_note(|| format!("CROSS_CMD={}", pre_build.join("\n")))?;
-            image = custom.image_name(target, metadata)?;
+            base_or_custom_image.name = custom.image_name(target, metadata)?;
         }
     }
-    Ok(image)
+    Ok(base_or_custom_image.name)
 }
 
-pub(crate) fn image_name(config: &Config, target: &Target) -> Result<String> {
+pub(crate) fn get_image(config: &Config, target: &Target) -> Result<PossibleImage> {
     if let Some(image) = config.image(target)? {
         return Ok(image);
     }
 
-    if !DOCKER_IMAGES.contains(&target.triple()) {
+    let compatible = PROVIDED_IMAGES
+        .iter()
+        .filter(|p| p.name == target.triple())
+        .collect::<Vec<_>>();
+
+    if compatible.is_empty() {
         eyre::bail!(
             "`cross` does not provide a Docker image for target {target}, \
                specify a custom image in `Cross.toml`."
         );
     }
 
+    if compatible.len() > 1 {
```

Comment:
This needs to be implemented, and should check the host platform to the image platform IMO. We should then be able to assert the length is 1.